physics/lagrange_equipotentials_body.hpp

```diff
@@ -0,0 +1,234 @@
+
+namespace principia {
+namespace physics {
+namespace _lagrange_equipotentials {
+namespace internal {
+
+using namespace principia::geometry::_barycentre_calculator;
+using namespace principia::geometry::_grassmann;
+using namespace principia::geometry::_plane;
+using namespace principia::integrators::_embedded_explicit_runge_kutta_integrator;  // NOLINT
+using namespace principia::integrators::_methods;
+using namespace principia::numerics::_global_optimization;
+using namespace principia::numerics::_root_finders;
+using namespace principia::physics::_rotating_pulsating_reference_frame;
+using namespace principia::quantities::_elementary_functions;
+using namespace principia::quantities::_named_quantities;
+using namespace principia::quantities::_quantities;
+using namespace principia::quantities::_si;
+
+template<typename Inertial, typename RotatingPulsating>
+LagrangeEquipotentials<Inertial, RotatingPulsating>::LagrangeEquipotentials(
+    not_null<Ephemeris<Inertial> const*> const ephemeris)
+    : ephemeris_(ephemeris) {}
+
+template<typename Inertial, typename RotatingPulsating>
+absl::StatusOr<typename LagrangeEquipotentials<Inertial, RotatingPulsating>::
+                   Equipotentials>
+LagrangeEquipotentials<Inertial, RotatingPulsating>::ComputeLines(
+    Parameters const& parameters) {
+  Equipotentials result;
+  Instant const& t = parameters.time;
+
+  RotatingPulsatingReferenceFrame<Inertial, RotatingPulsating> const
+      reference_frame(ephemeris_, parameters.primaries, parameters.secondaries);
+  auto const plane = Plane<RotatingPulsating>::OrthogonalTo(
+      Vector<double, RotatingPulsating>({0, 0, 1}));
+
+  auto const potential = [&reference_frame,
+                          &t](Position<RotatingPulsating> const& position) {
+    return reference_frame.GeometricPotential(t, position);
+  };
+  auto const gradient = [&reference_frame,
+                         &t](Position<RotatingPulsating> const& position) {
+    auto const acceleration = reference_frame.GeometricAcceleration(
+        t, {position, Velocity<RotatingPulsating>{}});
+    // Note the sign: the acceleration goes down the potential, we need the
+    // gradient which goes up.
+    return -Vector<Acceleration, RotatingPulsating>(
+        {acceleration.coordinates()[0],
+         acceleration.coordinates()[1],
+         Acceleration{}});
+  };
+
+  // The distance between the bodies is 1 m in the rotating-pulsating frame, and
+  // the origin is at their barycentre.  A 3 m box is large enough to encompass
+  // the L₄ and L₅ Lagrange points, as well as the whole ridge they sit on,
+  // regardless of the mass ratio.
+  const MultiLevelSingleLinkage<SpecificEnergy,
+                                Position<RotatingPulsating>,
+                                2>::Box box = {
+      .centre = RotatingPulsating::origin,
+      .vertices = {
+          Displacement<RotatingPulsating>({3 * Metre, 0 * Metre, 0 * Metre}),
```

Comment:
Let's gather all the constants (like `3 * Metre`) at the beginning of the body.  Some of them we might want to tune or move to the blueprint.